src/plugins/request/src/main/java/org/drftpd/request/master/Request.java

```diff
@@ -228,177 +274,198 @@ public CommandResponse doSITE_REQUEST(CommandRequest request) throws ImproperUsa
     }
 
     public CommandResponse doSITE_REQUESTS(CommandRequest request) throws ImproperUsageException {
+
+        // Command has no arguments
         if (request.hasArgument()) {
             throw new ImproperUsageException();
         }
 
-        Map<String, Object> env = new HashMap<>();
-        CommandResponse response = StandardCommandManager.genericResponse("RESPONSE_200_COMMAND_OK");
-        response.addComment(request.getSession().jprintf(_bundle, "requests.header", env, request.getUser()));
-        int i = 1;
-
-        User user = request.getSession().getUserNull(request.getUser());
-
-        SimpleDateFormat sdf = new SimpleDateFormat(_requestDateFormat);
-
-        try {
-
-                ArrayList<RequestsSort> ReqSort = new ArrayList<>();
+        DirectoryHandle requestDir = getRequestDirectory(request);
+        RequestData requests = getRequestData(requestDir);
 
-                for (DirectoryHandle dir : getRequestDirectory(request).getDirectories(user)) {
-                if (!dir.getName().startsWith(_requestPrefix)) {
-                    continue;
-                }
+        if (requests == null) {
+            return new CommandResponse(500, "Internal Server error occurred, stopping execution");
+        }
 
-                RequestParser parser = new RequestParser(dir.getName(), dir.getUsername());
+        CommandResponse response = StandardCommandManager.genericResponse("RESPONSE_200_COMMAND_OK");
+        response.addComment(request.getSession().jprintf(_bundle, "requests.header", request.getUser()));
 
-                ReqSort.add(new RequestsSort(parser.getRequestName(), dir.getUsername(), dir.getInode().getCreationTime()));
-                
-            }
+        SimpleDateFormat sdf = new SimpleDateFormat(getSettings().getRequestDateFormat());
 
-            Collections.sort(ReqSort);
-            for(RequestsSort rs:ReqSort){
+        int i = 1;
+        for (RequestEntry reqEntry : requests.getRequests()) {
+            Map<String, Object> env = new HashMap<>();
+            env.put("num", Integer.toString(i++));
+            env.put("request.name", reqEntry.getName());
+            env.put("request.user", reqEntry.getUser());
+            env.put("request.date", sdf.format(reqEntry.getCreationTime()));
```

Comment:
`reqEntry.getCreationTime()` seems to contain the correct epoch timestamp, but sdf is converting it to `1970-01-19 @ 18:59` so something is wrong here.

In previous code the input to sdf.,format was a Date object